src/sentry/api/bases/organizationmember.py

```diff
@@ -41,19 +52,27 @@ def convert_args(self, request: Request, organization_slug, member_id="me", *arg
             except OrganizationMember.DoesNotExist:
                 raise ResourceDoesNotExist
 
-            return (args, kwargs)
+            return args, kwargs
         else:
             raise ResourceDoesNotExist
 
-    def _get_member(self, request: Request, organization, member_id):
+    def _get_member(
+        self,
+        request: Request,
+        organization: Organization,
+        member_id: int | str,
+        invite_status: InviteStatus | None = None,
+    ) -> OrganizationMember:
+        args = []
+        kwargs = dict(organization=organization)
+
         if member_id == "me":
-            queryset = OrganizationMember.objects.filter(
-                organization=organization, user__id=request.user.id, user__is_active=True
-            )
+            kwargs.update(user__id=request.user.id, user__is_active=True)
         else:
-            queryset = OrganizationMember.objects.filter(
-                Q(user__is_active=True) | Q(user__isnull=True),
-                organization=organization,
-                id=member_id,
-            )
-        return queryset.select_related("user").get()
+            args.append(Q(user__is_active=True) | Q(user__isnull=True))
+            kwargs.update(id=member_id)
+
+        if invite_status:
+            kwargs.update(invite_status=invite_status.value)
+
+        return OrganizationMember.objects.filter(*args, **kwargs).select_related("user").get()
```

Comment:
do we use this pattern of `args` and `kwargs` anywhere else in the codebase for filtering? overall I find it harder to read.